datafusion/physical-expr/Cargo.toml

```diff
@@ -60,6 +60,7 @@ lazy_static = { version = "^1.4.0" }
 md-5 = { version = "^0.10.0", optional = true }
 num-traits = { version = "0.2", default-features = false }
 paste = "^1.0"
+petgraph = "0.6.2"
```

Comment:
I reviewed this crate and its dependencies: https://crates.io/crates/petgraph/reverse_dependencies and it looks fine to add to me.

Given we already have a dependency on parking_lot I don't think this is a net new dependency, and furthermore it seems widely used across the Rust ecosystem.